### PR TITLE
Wunused macros

### PR DESCRIPTION
**Pull request by andrei-sandor**
COMP: Fix unused macros related to MSVC and OS

This is to fix the -Wunused-macros generated by clang for macros related to MSVC and Microsoft.

For itkNumericTraitsTest.cxx, the _MSC_VER is added to respect the comment.

For itkTestDriver.cxx, ITK_TEST_DRIVER_PATH_SLASH / is related to non Windows machine and there is none in the code and hence it is removed. Now, for ITK_TEST_DRIVER_PATH_SLASH \\, it is not found in the entire code. This is the reason why it is removed.

------------------------------------------------------------------------------------------------

 COMP: Fix unused general macros

This is to fix the -Wunused-macros warning from clang. This is about fixing general macros used in the code.

In the commit, the macros are removed. However, I am wondering if we should comment them since someone might use them in the future.

------------------------------------------------------------------------------------------------

 COMP: Fixed unused legacy remove Macros

This is a commit to solve the Wunused-macros warning generated by clang related to ITK_LEGACY_REMOVE macro.

The change is to create an ifnded macro with it even if ITK_LEGACY_REMOVE is not used in the file. In itkConfigue.h, ITK_LEGACY_REMOVE is defined there if the code is in the legacy remove scenario or not. We need to keep ITK_LEGACY_REMOVE to create a flag that the code is legacy.

### CODE REVIEW

File: Modules/IO/BioRad/src/itkBioRadImageIO.cxx [hunk 1]
```diff
@@ -31,9 +31,6 @@
 #include "itkMakeUniqueForOverwrite.h"
 
 #define BIORAD_HEADER_LENGTH 76
-#define BIORAD_NOTE_LENGTH 96
```
Comment (dzenanz) [inline Modules/IO/BioRad/src/itkBioRadImageIO.cxx:34]: I would prefer to comment these out. While this code is unlikely to change, having this commented out is better than removed.

Comment (seanm) [reply Modules/IO/BioRad/src/itkBioRadImageIO.cxx:34]: For this file only, or all similar cases in this same commit?

Comment (dzenanz) [reply Modules/IO/BioRad/src/itkBioRadImageIO.cxx:34]: For all the changes in this commit.

File: Modules/IO/GIPL/src/itkGiplImageIO.cxx [hunk 2]
```diff
@@ -40,35 +40,34 @@ class GiplImageIOInternals
...
-#define OBLIQUE_TOMO 12
+// #define GIPL_C_SHORT 144
+// #define GIPL_C_INT 160
+// #define GIPL_C_FLOAT 192
+// #define GIPL_C_DOUBLE 193
+// #define GIPL_SURFACE 200
+// #define GIPL_POLYGON 201
+//
+// /*  ORIENTATION DEFINITIONS (flag1)  */
+//
```
Comment (seanm) [inline Modules/IO/GIPL/src/itkGiplImageIO.cxx:51]: These 3 lines don't need to have `//` added

File: Modules/Core/Common/test/itkSTLThreadTest.cxx [hunk 3]
```diff
@@ -15,8 +15,9 @@
  *  limitations under the License.
  *
  *=========================================================================*/
-
-#define ITK_LEGACY_TEST // so deprecation warnings are not triggered by this test
+#ifndef ITK_LEGACY_REMOVE
+#  define ITK_LEGACY_TEST
+#endif
```
Comment (seanm) [inline Modules/Core/Common/test/itkSTLThreadTest.cxx:20]: The explanatory comment should be kept.

File: Modules/IO/GIPL/src/itkGiplImageIO.cxx [hunk 4]
```diff
@@ -40,35 +40,34 @@ class GiplImageIOInternals
...
 
 /*  FORMAT DEFINITIONS  */
 
-#define FORMAT_GIPL 0
-#define FORMAT_GIPL_STRING "Gipl"
-#define FORMAT_MAYO 1
-#define FORMAT_MAYO_STRING "Mayo"
-#define FORMAT_NM_IGE 2
-#define FORMAT_NM_IGE_STRING "Starcam"
-
```
Comment (seanm) [inline Modules/IO/GIPL/src/itkGiplImageIO.cxx:71]: Sorry to be so nitpicky, but this empty line should be brought back I guess, as presumably these defines are grouped in some manner.

**Comment by seanm** [comment]
/azp.run ITK.Linux

**Review by seanm — APPROVED** [review]

**Review by hjmjohnson — APPROVED** [review]

**Pull request merged**